osu.Game.Rulesets.Tau/Allocation/MemoryPool.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace osu.Game.Rulesets.Tau.Allocation
+{
+    public class MemoryPool<T>
```

Comment:
Would probably be best to document this class, because I have no idea what the fuck is going on here. Maybe add some tests too?